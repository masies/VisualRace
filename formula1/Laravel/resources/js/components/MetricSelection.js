import React from 'react';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';

class MetricSelection extends React.Component {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <Grid container spacing={1} direction="column" >
        {
          Object.entries(this.props.metricButtons).map((metric) =>
            <Grid container key={metric[0]} direction="column" align="center" style={{ marginBottom: "7px" }}>
              <Button id={metric[0]} onClick={() => this.props.handler(metric[0])} fullWidth={true} variant={metric[1].variant} color="primary">
                {metric[0]}
              </Button>
            </Grid>
          )
        }
      </Grid>
    )
  }
}

export default MetricSelection;