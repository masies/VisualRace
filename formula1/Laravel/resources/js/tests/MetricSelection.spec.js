import React from 'react';
import { shallow, mount } from 'enzyme';
import MetricSelection from '../components/MetricSelection';


let metricButtons = {
  speed: {
    show: true,
    variant: 'contained'
  }
}


let metricButtons2 = {
  throttle: {
    show: true,
    variant: 'contained'
  }
}

const toggleButton = jest.fn()

let wrapped = mount(<MetricSelection handler={toggleButton} metricButtons={metricButtons} />);
let wrapped_shallow = shallow(<MetricSelection handler={toggleButton} metricButtons={metricButtons} />);

describe('MetricSelection', () => {
  it('should render the MetricSelection Component correctly', () => {
    expect(wrapped_shallow).toMatchSnapshot();
  });

  it('button click should toggle metric selection', () => {

    wrapped.find('button').simulate('click')
    expect(toggleButton).toHaveBeenCalled()
  })

  it('button text should match metric name', () => {
    expect(wrapped.find('button').text()).toEqual('speed')
  })

  it('should change buttons state when props change', () => {
    wrapped.setProps({ metricButtons: metricButtons2 })
    expect(wrapped.find('button').text()).toEqual('throttle')

  })

});