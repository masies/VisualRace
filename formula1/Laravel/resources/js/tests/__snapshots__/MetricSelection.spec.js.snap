// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetricSelection should render the MetricSelection Component correctly 1`] = `ShallowWrapper {}`;
